manifests/port.pp

```diff
@@ -46,11 +46,13 @@
   if $protocol {
     validate_re($protocol, ['^tcp6?$', '^udp6?$'])
     $protocol_switch="-p ${protocol} "
+    $port_exec_command = "add_${context}_${port}_${protocol}"
   } else {
     $protocol_switch=''
+    $port_exec_command = "add_${context}_${port}"
   }
 
-  exec { "add_${context}_${port}":
+  exec { '$port_exec_command':
```

Comment:
Your string isn't being interpolated. Single quotes. Just remove the quotes, imo.